x-pack/test/apm_api_integration/tests/time_range_metadata/many_apm_server_versions.spec.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import expect from '@kbn/expect';
+import { apm, ApmFields, timerange } from '@kbn/apm-synthtrace-client';
+import { Transform } from 'stream';
+import moment from 'moment';
+import { ApmSynthtraceEsClient } from '@kbn/apm-synthtrace';
+import {
+  TRANSACTION_DURATION_HISTOGRAM,
+  TRANSACTION_DURATION_SUMMARY,
+} from '@kbn/apm-plugin/common/es_fields/apm';
+import { ApmDocumentType } from '@kbn/apm-plugin/common/document_type';
+import { RollupInterval } from '@kbn/apm-plugin/common/rollup';
+import { LatencyAggregationType } from '@kbn/apm-plugin/common/latency_aggregation_types';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { ApmApiClient } from '../../common/config';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+  const apmApiClient = getService('apmApiClient');
+  const synthtrace = getService('synthtraceEsClient');
+  const es = getService('es');
+
+  const baseTime = new Date('2023-10-01T00:00:00.000Z').getTime();
+  const startLegacy = moment(baseTime).add(0, 'minutes');
+  const start = moment(baseTime).add(5, 'minutes');
+  const end = moment(baseTime).add(10, 'minutes');
+
+  registry.when(
+    'Time range metadata when there are multiple APM Server versions',
+    { config: 'basic', archives: [] },
+    () => {
+      describe('when ingesting traces from APM Server with different versions', () => {
+        before(async () => {
+          await synthtrace.clean();
```

Comment:
Why do we need a clean even before starting running the test ?

Reply:
You can actually restructure your before/after clause something on these lines where i was generating data with and without summary fields

https://github.com/elastic/kibana/blob/d4402886a12e6c429d72f441ee988b384645fd57/x-pack/test/apm_api_integration/tests/time_range_metadata/time_range_metadata.spec.ts#L85-L100

Reply:
I've updated the `index` method to now take a list of transforms. That means you no longer have to restore the pipeline manually, like you did in time_range_metadata. Take a look and LMK what you think

Reply:
TBH, Initially i was not convinced with this idea to modify the Synth Client code given we are doing this due to a specific test, but after discussing with @dgieselaar, it makes sense to do it like this.

Initially the test i wrote was the only case. And now you have added an additional test which requires a transform. 
This means in future we might have more such use cases coming, thus, Synthtrace Client should be enhanced to handle passing additional transforms.

Thank you @sqren for adding this 👍🏼